### DISCUSSION TITLE
Triggering a workflow when PR "combined status" is updated

### DISCUSSION THREAD

**Discussion by boegel** [community/community#25547] [Actions]
<p>I’m trying to trigger a workflow when the “combined status” of a pull request is updated. I would like to use this for a workflow that notifies the PR author of failing tests in a PR, or to compose a workflow that auto-merges a PR when all requirements are met (all tests pass, labels/milestone set, etc.).</p>
<p>I tried achieves this using the check_run, check_suite and status events, but none of those seem to be triggering anything…</p>
<p>Other events in the same workflow (e.g. issue_comment) are working as expected.</p>
<p>A previous post was asking a similar question (<a href="https://github.community/t/how-to-trigger-workflow-when-all-checks-pass-for-pull-request-combined-status/16483">https://github.community/t5/GitHub-Actions/How-to-trigger-workflow-when-all-checks-pass-for-pull-request/m-p/33944#M1726</a>), but the answer didn’t help me at all…</p>
<p>Am I overlooking something obvious?</p>
<p>Test workflow:</p>
<pre><code>name: test
on:
  check_run:
    types: [completed]
  check_suite:
    types: [completed]
  issue_comment:
    types: [created]
  status: {}
jobs:
  ...
</code></pre>
<p>(see also <a href="https://github.com/boegel/test-github-actions/blob/master/.github/workflows/test.yml" rel="nofollow noopener">https://github.com/boegel/test-github-actions/blob/master/.github/workflows/test.yml</a>)</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/boegel">@boegel</a>,</p>
<p>Thank you for being here and our apologies for the delayed response. I believe you are running into a known limitation:</p>
<p><em>&gt; An action in a workflow run can’t trigger a new workflow run. For example, if an action pushes code using the repository’s GITHUB_TOKEN, a new workflow will not run even when the repository contains a workflow configured to run when push events occur.</em></p>

**Comment by WyriHaximus** [comment]
<p>Hey <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> ,</p>
<p>Thank you for your response. I’m also running into this, is there a planned fix for this? Or do we, assuming I read your quote right, have to use a token from <a href="https://github.com/settings/tokens" rel="nofollow noopener">https://github.com/settings/tokens</a> and let the action pose as us. Will it trigger then next action then?</p>

**Comment by WyriHaximus** [comment, marked as answer]
<blockquote>
<p><a class="mention" href="/wyrihaximus">@wyrihaximus</a> wrote:</p>
</blockquote>
<blockquote>
<p>Hey <a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> ,</p>
</blockquote>
<blockquote>
<p> </p>
</blockquote>
<blockquote>
<p>Thank you for your response. I’m also running into this, is there a planned fix for this? Or do we, assuming I read your quote right, have to use a token from <a href="https://github.com/settings/tokens" rel="nofollow noopener">https://github.com/settings/tokens</a> and let the action pose as us. Will it trigger then next action then?</p>
</blockquote>
<p>Ok so I can confirm that this works. Have something set up at <a href="https://github.com/WyriHaximus/php-fake-php-version/tree/master/.github/workflows" rel="nofollow noopener">https://github.com/WyriHaximus/php-fake-php-version/tree/master/.github/workflows</a> that will create a PR (with milestone and labels), auto merge it when all checks are green, close the milestone, and create a release. Each step is a workflow and triggers another workflow. They didn’t trigger with the default GITHUB_TOKEN, but they do when you create a custom token.</p>
